lib/game.rb

```diff
@@ -0,0 +1,170 @@
+require 'pry'
+
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  @@WIN_COMBINATIONS = [
+  [0, 1, 2],
+  [3, 4, 5],
+  [6, 7, 8],
+  [0, 3, 6],
+  [1, 4, 7],
+  [2, 5, 8],
+  [0, 4, 8],
+  [6, 4, 2]]
+
+  WIN_COMBINATIONS = @@WIN_COMBINATIONS # This is needed to pass a test but violates DRY. Check solution for more compact and elegant approach.
+
+  def self.win_combinations
+    @@WIN_COMBINATIONS
+  end
+
+  def initialize(player_1=Players::Human.new('X'), player_2=Players::Human.new('O'), board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def turn_count
+    count = 0
+    @board.cells.each do | c |
+      if c == 'X' || c == 'O'
+        count += 1
+      end
+    end
+    @player_1.turn_count = count # Could refactor turn_count as a class variable within Player for DRY.
+    @player_2.turn_count = count
+    count
+  end
+
+  def current_player
+    if turn_count.even?
+      @player_1
+    else
+      @player_2
+    end
+  end
+
+  def won?
+    @@WIN_COMBINATIONS.each do |wc|
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```